os/sys/stack-check.c

```diff
@@ -73,7 +73,7 @@ extern int _stack_origin;
 void
 stack_check_init(void)
 {
-  uint8_t *p;
+  volatile uint8_t *p;
```

Comment:
Please add a comment describing the optimization issue.

Reply:
Done. :)